scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -658,6 +658,16 @@ protected boolean createExecution(NodeSet nodeSet, Node node, InternalJob job, I
 
                     tlogger.debug(task.getId(), "deploying");
 
+                    // Dynamically adjust the start-task-timeout according to the number dependency tasks in a merge.
+                    // above 500 parent tasks, it is worth adjusting.
+                    if (taskDescriptor.getParents().size() > 500) {
+                        DOTASK_ACTION_TIMEOUT = (int) (taskDescriptor.getParents().size() / 500.0 *
+                                                       PASchedulerProperties.SCHEDULER_STARTTASK_TIMEOUT.getValueAsInt());
+                    } else {
+                        // reset the DOTASK_ACTION_TIMEOUT to its default value otherwise.
+                        DOTASK_ACTION_TIMEOUT = PASchedulerProperties.SCHEDULER_STARTTASK_TIMEOUT.getValueAsInt();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)

Reply:
This is legacy code and cannot simply we cannot simply transform the method into static